onchain/rollups/contracts/library/LibInput.sol

```diff
@@ -12,36 +12,57 @@ library LibInput {
     /// @notice Raised when input is larger than the machine limit.
     error InputSizeExceedsLimit();
 
-    /// @notice Summarize input data in a single hash.
+    /// @notice Encode input with EvmInput signature
     /// @param sender `msg.sender`
     /// @param blockNumber `block.number`
     /// @param blockTimestamp `block.timestamp`
-    /// @param input The input blob
-    /// @param inputIndex The index of the input in the input box
-    /// @return The input hash
-    function computeInputHash(
+    /// @param index The index of the input in the input box
+    /// @param payload The input payload
+    /// @return The encoded EVM input
+    function encodeEvmInput(
         address sender,
         uint256 blockNumber,
         uint256 blockTimestamp,
-        bytes calldata input,
-        uint256 inputIndex
-    ) internal pure returns (bytes32) {
-        if (input.length > CanonicalMachine.INPUT_MAX_SIZE) {
-            revert InputSizeExceedsLimit();
-        }
-
-        bytes32 keccakMetadata = keccak256(
-            abi.encode(
+        uint256 index,
+        bytes memory payload
```

Comment:
Please make this `calldata`, for gas cost reasons. In the tests, you can create a contract that receives `calldata` which you can call with a payload on `memory`.

Reply:
I'm curious, how would other contracts (besides the test contract) actually call this function then? Do they also create an intermediary contract or though low level calls to call this function?